### ISSUE TITLE
[Feature]: spm additional query parameter required for prometheus

### ISSUE THREAD

**Issue by rbaumgar** [jaegertracing/jaeger#4221]
### Requirement

I am using a global prometheus for spm where I need to add a specific query parameter to get data.
Currently no parameter is available to define an additional query parameter.
I only found --query.base-path which does not fit.

### Problem

spm sends only e.g. "/api/v1/query?query=up", but I need to send "/api/v1/query?query=up&mykey=myvalue"
otherwise I don't get any data.

### Proposal

please provide another parameter, eg. --query.additional-parameter.

**Comment by hkailantzis** [comment]
That would be great, as in my case we're using Thanos, and query parameter is a hard requirement for us (like the namespace), and can't alter it this requirement. For the Bearer token I use the ```---prometheus.token-file cli flag```, which is basically a secret token generated by an appopriate SA (for all-in-one), and I'm mounting is using this:

```
extraSecretMounts:
    - name: prometheus-token
      mountPath: /tls
      secretName: jaeger-all-in-one-token
      readOnly: true
```
this plus RoleBinding of ClusterRole type:

```
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: jaeger
  namespace: default
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: <ClusterRoleName>
subjects:
- kind: ServiceAccount
  name: jaeger-all-in-one
  namespace: default
  ```
but I'm getting an error which I can't make anything out of it, so I'm assuming is because of the missing required namespace parameter..., but not 100% sure...

```
msg":"HTTP handler, Internal Server Error","error":"failed executing metrics query: bad_response: readObjectStart: expect { or n, but found B, error found in #1 byte of ...|Bad Request|..., bigger context ...|Bad Request. The request or configuration is malfor|..."
```

cc @rbaumgar

**Comment by rbaumgar** [comment]
@hkailantzis I don't know what is required in your environment.
In OpenShift you have two ports available
- 9091, requires a cluster role. access to all namespace data
- 9092, requires a namespace parameter where the token has access to
https://www.redhat.com/en/blog/thanos-querier-versus-thanos-querier

**Comment by hkailantzis** [comment]
@rbaumgar, is the 9092 case actually, where a namespace parameter is required...which is not supported by jaeger SPM as the moment right ?

**Comment by rbaumgar** [comment]
@hkailantzis correct! I don't want to use a cluster role for SPM...

**Comment by prakrit55** [comment]
@yurishkuro, I want to try it out

**Comment by yurishkuro** [comment]
@akstron we pass `api.Client` to the api.v1 which is used for executing the actual HTTP requests. The client is a very small API that is easy to wrap in a decorator if we want to add some extra parameters to the URL.

**Comment by akstron** [comment]
I see, that makes sense. I will start working on it. Thanks!

**Comment by yurishkuro** [comment]
@akstron we should only do this in Jaeger v2 (i.e. don't both with v1 CLI flags for these headers).

**Comment by yurishkuro** [comment]
This will be available in v2.2.0 via `extra_query_parameters` field in the Prometheus backend config.
